torchdata/datapipes/iter/load/s3io.py

```diff
@@ -108,20 +120,32 @@ class S3FileLoaderIterDataPipe(IterDataPipe[Tuple[str, StreamWrapper]]):
         multi_part_download: flag to split each chunk into small packets and download those packets in parallel (enabled by default)
 
     Example:
-        >>> from torchdata.datapipes.iter import IterableWrapper, S3FileLoader
-        >>> dp_s3_urls = IterableWrapper(['s3://bucket-name/folder/', ...]).list_files_by_s3()
+
+    .. testsetup::
+
+        from torchdata.datapipes.iter import S3FileLoader
+
+        S3FileLoader.__iter__ = lambda self: iter([])
```

Comment:
Ditto

Reply:
changed to patching in testsetup and cleaning up the patch in testcleanup